### ISSUE TITLE
'Recent' tab not displaying all recent objects.

### ISSUE THREAD

**Issue by panditthecoder** [objecthub-io/ObjectHub#244]
The recent tab is not showing all recent objects. 
I opened a new browser (cleared cache, cookies the works to ensure that this was a clean browser).
used only Micro_Apps.
- created microapp called Object_No_1. added html, pinned it. checked in the 'recent' tab - object showed up. checked in 'Pins' Tab - object showed up. 
All wel and good so far - did not go to the JS, CSS, Preview or Deploy tad
- created a second object called Object_No_2. added html, Pinned it. Checked in the 'recent' tab - it did not show up there. checked in pin tab - did not show up there as well. So that was a bug. (again did not click on the js, css, preview or deploy tab) - so this was a bug - wanted to explore a bit more to ensure that the fact that the name has the first 10 char common was not the cause - should not be but i wanted to be sure.
- created a 3rd microapp called Third_object_noticeTheChangeInName. added HTML. Checked in the 'recent' tab - it did not show up there - neither did Object_No_2. Hence the bug.
Image of browser when i go to recent tab - notice, I am editing the "Third_object_noticeTheChangeInName" object so by rights, it should be in the recent tab
![object_io_not_showing_in_recent_or_pin_tab](https://user-images.githubusercontent.com/29138377/27544361-869071fc-5aaa-11e7-8255-907ed1e57c4b.JPG)

Image of the 'My Objects' tab. here you will see the list of objects created recently - Object_No_1, Object_No_2 and Third_object_noticeTheChangeInName in that sequence. however, (from the previous image) neither Object_No_2 not Third_object_noticeTheChangeInName showed up there.
![object_sequence_of_creation](https://user-images.githubusercontent.com/29138377/27544522-071daca4-5aab-11e7-992f-0d1c0a051718.JPG)

Signed out and signed back into my account and volla all the objects show up in the recent tab. 
![object_recent_tab_after_signoff_and_sign_back_in](https://user-images.githubusercontent.com/29138377/27544562-256a912c-5aab-11e7-8572-a51669704904.JPG)

**Comment by panditthecoder** [comment]
while fixing it, can you also please check for this sequence
- create an object (object 1) just add some code - see if it shows up in recent tab.
- create object 2 - see if it shows up in recent tab - if yes, go back to my objects tab and delete object 2
- create object 3 - do not go to the recent or pin tab - delete it
- create object 4 - go to the recent tab and see what shows up there. by rights only object 4 and 1 should show up - if anything more or less, then that is a bug
- create a new object called whatever you had called Object 2 (which you had deleted) - Use the same name exactly. see how that fares in the recent tab. 

the scenario (though it may look weird in the sequence I provided) is
- I want to create a complete app+service. so I start coding one object. I then start on another object and that does not do the stuff I want it to do so i delete the object. I then create another html page (or microapp). I then realize that the initial object I had created in step 2 is required and so i create it again - and this recreation may happen with multiple objects being created, edited or destroyed in the interim.

**Comment by objecthub-io** [comment]
Thank you for reporting this issue.

I can confirm that sometimes an object might not show up in the recent tab until after a reload. This is a bug and we'll fix it.

Could you clarify your second comment about the sequence and/or maybe create a separate issue for it?

**Comment by panditthecoder** [comment]
NM the second comment. let me know once the fix is done and i will test it out and log a fresh bug if reqd

**Comment by objecthub-io** [comment]
This issue should be fixed now.

**Comment by objecthub-io** [comment]
Closing fixed issues.
